src/unitxt/artifact.py

```diff
@@ -237,6 +244,30 @@ def from_dict(cls, d, overwrite_args=None):
     @classmethod
     def load(cls, path, artifact_identifier=None, overwrite_args=None):
```

Comment:
I suggest we override the load method in the derived ArtifactLink class, and not condition the behavior based on the existing the  attribute.